pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
```

Comment:
These are generic details core to the platform (image name).  Unfilled
fields can default to nothing.  Having multiple structs for generic
fields isn't useful.